### DISCUSSION TITLE
FEATURE: Disable Sticky Keys (follow-on to #1)

### DISCUSSION THREAD

**Discussion by tildebyte** [cafali/SnapKey#74] [New Features & Ideas]
FEATURE: Disable Sticky Keys (follow-on to #1)

I personally am not interested in this particular feature, and I'd like to be able to disable it.

Further than that, though; this seems to be the feature which is most upsetting to CS players (and IIUC, to Valve as well, but I have no actual insight there).

Letting us disable Sticky Keys might be enough to lift the ban 🤷

Full disclosure: I don't play CS, don't care about CS, don't *actually* care about CS players, HOWEVER - there *is* such a thing as bad publicity, and drawing the wrath of one of the most important companies in PC gaming as a months-old open source project doesn't do *anyone* any good. Just my 2¢.

**Comment by cafali** [comment]
What feature exactly? This is how the original 'Snap Tap' works.

The option to disable Snapkeys' functionality is already implemented.

**Comment by minteeaa** [comment]
alongside the already implemented ability to disable SnapKey as a whole, I highly doubt the "wrath" we draw as a small open source project in comparison to the keyboard manufacturers that are releasing this as a hardware feature is enough to classify the ban as more on us than it is on the multi-million dollar companies that threw this feature haphazardly into their products.

simply being able to "turn off" the feature won't be the golden ticket for Valve to lift anything - if it were that simple you could also just say "well don't use it" to the people that have a Wooting or Huntsman.

comes down to the skill gap between using it and not using it - not to the accessibility that we should have to this feature

**Comment by tildebyte** [comment]
It's possible (likely) that I don't understand the difference between "Sticky Keys" and the core SnapKey functionality...

No offense meant, but your commit messages suck, so I'm completely unable to find where "Sticky Keys" was introduced (and I'm not going to search through 160+ commits to find it), except that it appears in these release notes https://github.com/cafali/SnapKey/releases/tag/v1.1.4 - to wit:

> Introducing Sticky Keys, a feature that remembers the pressed state of the initially held button even after pressing another button

so perhaps you can see why I thought it was additional functionality beyond the core.

**Comment by minteeaa** [comment]
no offense taken - just meant my comment as a general clear up for that specific issue of a misunderstanding between `Sticky Keys` and core functionality; when referring to the feature of "Snap Tap" as a whole, "Sticky Keys" is usually just under that umbrella, so it makes sense. it's not like the actual feature from Razer and Wooting makes this distinction very well either

point I was trying to make is regardless of being able to disable a specific part of SnapKey or any other SOCD-like functionality, it will always be umbrella'd into the same black-or-white image of "cheating or not." disabling Sticky Keys in specific doesn't do anything to help that decision, the issue is Snap Tap/SOCD as a whole that Valve is trying to combat (and obviously really sucks at, per usual)

**Comment by tildebyte** [comment]
Given #76, it might be a good idea to include

> when referring to the feature of "Snap Tap" as a whole, "Sticky Keys" is usually just under that umbrella, so it makes sense. it's not like the actual feature from Razer and Wooting makes this distinction very well either

in the docs somewhere
